subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -303,11 +307,50 @@ private GroovyJavaJointCompileSpec createSpec() {
             GFileUtils.mkdirs(dir);
             spec.getGroovyCompileOptions().setStubDir(dir);
         }
-        spec.getCompileOptions().getForkOptions().setExecutable(Jvm.current().getJavaExecutable().getAbsolutePath());
+
+        if (javaLauncher.isPresent()) {
+            applyToolchain(spec.getCompileOptions().getForkOptions());
+        } else {
+            spec.getCompileOptions().getForkOptions().setExecutable(Jvm.current().getJavaExecutable().getAbsolutePath());
+        }
 
         return spec;
     }
 
+    private void configureCompatibilityOptions(DefaultGroovyJavaJointCompileSpec spec) {
```

Comment:
Should we move this to AbstractCompile?

Reply:
Maybe. I would do it if `ScalaCompile` ends up needing it as well